code/lib/preview-api/src/modules/store/csf/composeConfigs.test.ts

```diff
@@ -42,20 +42,52 @@ describe('composeConfigs', () => {
     });
   });
 
+  it('composes export defaults', () => {
+    expect(
+      composeConfigs([
+        {
+          default: {
+            parameters: { obj: { a: '1', b: '1' } },
+          },
+        },
+        {
+          default: {
+            parameters: { obj: { a: '2', c: '2' } },
+          },
+        },
+      ])
+    ).toEqual({
+      parameters: { obj: { a: '2', b: '1', c: '2' } },
+      decorators: [],
+      args: {},
+      argsEnhancers: [],
+      argTypes: {},
+      argTypesEnhancers: [],
+      globals: {},
+      globalTypes: {},
+      loaders: [],
+      runStep: expect.any(Function),
+    });
+  });
+
   it('overrides object fields by key', () => {
     expect(
       composeConfigs([
         {
-          args: { x: '1', y: '1', obj: { a: '1', b: '1' } },
-          argTypes: { x: '1', y: '1', obj: { a: '1', b: '1' } },
-          globals: { x: '1', y: '1', obj: { a: '1', b: '1' } },
-          globalTypes: { x: '1', y: '1', obj: { a: '1', b: '1' } },
+          default: {
+            args: { x: '1', y: '1', obj: { a: '1', b: '1' } },
+            argTypes: { x: '1', y: '1', obj: { a: '1', b: '1' } },
+            globals: { x: '1', y: '1', obj: { a: '1', b: '1' } },
+            globalTypes: { x: '1', y: '1', obj: { a: '1', b: '1' } },
+          },
         },
         {
-          args: { x: '2', z: '2', obj: { a: '2', c: '2' } },
-          argTypes: { x: '2', z: '2', obj: { a: '2', c: '2' } },
-          globals: { x: '2', z: '2', obj: { a: '2', c: '2' } },
-          globalTypes: { x: '2', z: '2', obj: { a: '2', c: '2' } },
+          default: {
```

Comment:
Could you add a case where this is mixed? We are composing configs that could come from users, addons, presets and frameworks, which will definitely be mixed and the composing of it all should work correctly

Reply:
Will do once CI finishes & merge

Reply:
It's all good – I tested that and added a telescoping PR for after we merge this: https://github.com/storybookjs/storybook/pull/21230